### ISSUE TITLE
ReleaseNotesGrabber Override HttpClient Global Injection

### ISSUE THREAD

**Issue by iamjplant** [NetSparkleUpdater/NetSparkle#388]
While it is nice that I can now override the CreateHttpClient in the ReleaseNotesGrabber, I have no particular way to do anything meaningful with that, as I cannot control when a ReleaseNotesGrabber is new'd up.  It would be nice if I could inject my own ReleaseNotesGrabber the way I can with a IAppCastDataDownloader.

My greater need for this is because I may desire to set my own User Agent.  My update files are in S3 buckets and there is currently a security restriction in our setup requiring User Agents to be set.

**Comment by iamjplant** [comment]
Probably related to #352

**Comment by Deadpikle** [comment]
Hi @iamjplant,

Yeeeeahhh......the easiest way to do that right now is to create your own UIFactory, override `CreateUpdateAvailableWindow`, and then set the `ReleaseNotesGrabber` property on the window's data context (or similar, depending on which UI you are using):

```csharp

    class MyUIFactory : UIFactory
    {
        public override IUpdateAvailable CreateUpdateAvailableWindow(SparkleUpdater sparkle, List<AppCastItem> updates, bool isUpdateAlreadyDownloaded = false)
        {
            var window = base.CreateUpdateAvailableWindow(sparkle, updates, isUpdateAlreadyDownloaded);
            if (window is UpdateAvailableWindow uaw && uaw.DataContext is UpdateAvailableWindowViewModel uawvm)
            {
                //uawvm.ReleaseNotesGrabber = new ReleaseNotesGrabber(...)
            }
            return window;
        }
    }
```

But that's a pain, I would agree. 🤪 It would be easier if there was a direct way to access/override this object from the `UIFactory` itself, yes.

Thank you for filing!

**Comment by Deadpikle** [comment]
You can test this out in 2.1.4-preview20221106001, rolling out now via CI/CD. Set `UIFactory.ReleaseNotesGrabberOverride`. Please let me know if that helps!

**Comment by iamjplant** [comment]
Yes, that looks like it will fit the bill.  Very cool.

**Comment by iamjplant** [comment]
When do you plan to have 2.1.4 deployed?  Thanks!

**Comment by Deadpikle** [comment]
Released in v2.2.0
